### PR TITLE
Correct filter name in voucher filters for old dashboard

### LINKED ISSUE #4382: Incorrect filter name in voucher filters for old dashboard
In filters for old dashboard we use:

```
DISCOUNT_TYPE_CHOICES = (
    ("fixed", pgettext_lazy("Sale type filter choice", "USD")),
    ("percentage", pgettext_lazy("Sale type filter choice", "%")),
)
```

In model, we store this information in `discount_value_type` field. In `discount_type` field, we have information about voucher type e.g. `ENTIRE_ORDER`, `SPECIFIC_PRODUCT`, `SHIPPING`

### PR DESCRIPTION
**Pull request by krustocious27**
Added voucher type choices to voucher filter choices. This attempts to address issue #4382. Kept variable naming from models.py.

I want to merge this change because...

First PR for Hacktoberfest! It closes #4382!

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for c67703d1b5328cad10985a289dfa7525fc536833 (krustocious27/saleor @ correct-filter-name-in-voucher-filters-for-old-dashboard)
Base comparison is 041f2aeab90b132abbbde9c505355413c3e4d601.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         14	         14	              0
  complete checkout                          	          6	          6	              0
  create checkout                            	         48	         48	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	         13	         13	              2

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/dashboard/discount/filters.py [hunk 1]
```diff
@@ -34,11 +34,18 @@
...
+DISCOUNT_VALUE_TYPE_CHOICES = (
     ("fixed", pgettext_lazy("Sale type filter choice", "USD")),
     ("percentage", pgettext_lazy("Sale type filter choice", "%")),
 )
 
+VOUCHER_TYPE_CHOICES = (
+    ("shipping", pgettext_lazy("Voucher type filter choice", "Shipping")),
+    ("entire_order", pgettext_lazy("Voucher type filter choice", "Entire Order")),
+    ("specific_product", pgettext_lazy("Voucher type filter choice",
+                                       "Specific Product")),
```
Comment (fowczarek) [inline saleor/dashboard/discount/filters.py]: Please fix black.

File: saleor/dashboard/discount/filters.py [hunk 2]
```diff
@@ -34,11 +34,22 @@
...
 )
 
+VOUCHER_TYPE_CHOICES = (
+    ("shipping", pgettext_lazy("Voucher type filter choice", "Shipping")),
+    ("entire_order", pgettext_lazy("Voucher type filter choice", "Entire Order")),
+    (
+        "specific_product",
+        pgettext_lazy("Voucher type filter choice", "Specific Product"),
+            "Voucher type filter choice", "Specific Product"
+        )
```
Comment (fowczarek) [inline saleor/dashboard/discount/filters.py]: You should remove this.

**Comment by krustocious27** [comment]
Thank you for the python spacing help @fowczarek !

**Comment by maarcingebala** [comment]
Hey @krustocious27, thanks a lot for your contribution! It's already November, but the PR was opened when Hacktoberfest was still on and we will send you a gift! Please send me your shipping address and preferred socks size: 40-43 or 44-46.

**Comment by maarcingebala** [comment]
My email address is: marcin@mirumee.com

**Comment by krustocious27** [comment]
Hi @maarcingebala, sent you an email. Thank you!

**Pull request merged**